lib/using_restricted_array.rb

```diff
@@ -9,53 +11,109 @@
 # Time complexity: ?
 # Space complexity: ?
 def length(array)
-  raise NotImplementedError
+  # array[index]
+  i = 0
+  i += 1 until array[i].nil?
+
+  i
+  # raise NotImplementedError
 end
 
 # Prints each integer values in the array
 # Time complexity: ?
 # Space complexity: ?
 def print_array(array)
-  raise NotImplementedError
+  # raise NotImplementedError
+  i = 0
+  until array[i].nil?
+    print array[i]
+    i += 1
+  end
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 # Time complexity: ?
 # Space complexity: ?
-def search(array, length, value_to_find)
-  raise NotImplementedError
-end
+def search(array, _length, value_to_find)
+    # raise NotImplementedError
+    i = 0
+    until array[i] == nil
+      if array[i] == value_to_find
+        return true
+      else
+        return false
+      end
+    end
```

Comment:
This doesn't work because if the 1st element is **not** the item we're looking for it returns false and doesn't check the remainder of the array.

```suggestion
    until array[i] == nil
      if array[i] == value_to_find
        return true
      end
      i += 1
    end
    return false
```